Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md

```diff
@@ -14,60 +14,72 @@ In the following section, we list some common root detection methods you'll enco
 
-SafetyNet is an Android API that provides a set of services and creates profiles of devices according to software and hardware information. This profile is then compared to a list of accepted device models that have passed Android compatibility testing. Google [recommends](https://developers.google.com/android/reference/com/google/android/gms/safetynet/SafetyNet "SafetyNet Documentation") using the feature as "an additional in-depth defense signal as part of an anti-abuse system".
+**Google Play Integrity provides the following protections:**
 
-How exactly SafetyNet works is not well documented and may change at any time. When you call this API, SafetyNet downloads a binary package containing the device validation code provided from Google, and the code is then dynamically executed via reflection. An [analysis by John Kozyrakis](https://koz.io/inside-safetynet/ "SafetyNet: Google's tamper detection for Android") showed that SafetyNet also attempts to detect whether the device is rooted, but exactly how that's determined is unclear.
+Verification of genuine Android device: verifies that the app is running on a legitimate Android device.  
+User license validation: Indicates whether the user installed or paid for the app or game via the Google Play Store.  
+Unmodified binary verification: determines if the app is interacting with the original binary recognized by Google Play.
 
-To use the API, an app may call the `SafetyNetApi.attest` method (which returns a JWS message with the _Attestation Result_) and then check the following fields:
+The API provides four macro categories of information to help the security team make a decision. These categories include:
 
-- `ctsProfileMatch`: If 'true', the device profile matches one of Google's listed devices.
-- `basicIntegrity`: If 'true', the device running the app likely hasn't been tampered with.
-- `nonces`: To match the response to its request.
-- `timestampMs`: To check how much time has passed since you made the request and you got the response. A delayed response may suggest suspicious activity.
-- `apkPackageName`, `apkCertificateDigestSha256`, `apkDigestSha256`: Provide information about the APK, which is used to verify the identity of the calling app. These parameters are absent if the API cannot reliably determine the APK information.
+1.Request Details: In this section, details are obtained about the app package that requested the integrity check, including  its format (my.package.com), a base64-encoded ID provided by the developer to establish a link between the request and the integrity certificate, and the time of the request execution in milliseconds.

```

Comment:
Consider putting **Request Details** in bold, and apply the same style with points `2`, `3`, and `4`.

Reply:
Also please add a space after the dot, like this `1. `. If not, it won't be recognized as a list item.